TheOracle2/GameObjects/Die.cs

```diff
@@ -1,34 +1,57 @@
 namespace TheOracle2.GameObjects;
 
-public class Die
```

Comment:
This should probably be the new `record` type and not a class. I'm not quite sure what that does to the `IComparable<int>` though, it probably just needs to stay in the inheritance chain.